main.go

```diff
@@ -64,19 +73,48 @@ func extractSemVersionFromChannel(channel, prefix string) (*semver.Version, erro
 	return semver.NewVersion(trimmed)
 }
 
-// dropPatch returns a semver.Version containing only the major.minor parts.
-func dropPatch(v *semver.Version) (*semver.Version, error) {
-	newVersionStr := fmt.Sprintf("%v.%v", v.Major(), v.Minor())
-	return semver.NewVersion(newVersionStr)
+// splitFastVersion splits the input string into a prefix (including the hyphen)
+// and the version part. It assumes that the input always contains a hyphen.
+func splitChannel(channel string) (string, string, error) {
+	idx := strings.Index(channel, "-")
+	// If the hyphen is not found, return an empty prefix and the original input as version.
+	if idx == -1 {
+		return "", channel, fmt.Errorf("invalid channel format: %s", channel)
+	}
+	prefix := channel[:idx+1]
+	version := channel[idx+1:]
+	return prefix, version, nil
 }
 
 // discoverReleases discovers new releases from the startChannel and minimum acceptable version.
 // It returns a ReleasesByChannel, with keys as full channel names.
-func discoverReleases(client *http.Client, startChannel string, minVer, minChannelVer *semver.Version, prefixes []string, arch string) (ReleasesByChannel, error) {
+func discoverReleases(client *http.Client, startChannels []string, arch string) (ReleasesByChannel, error) {
+	var prefixes []string
+	var minVersion *semver.Version
+	var queue []string
+	queued := map[string]bool{}
+
+	for _, startChannel := range startChannels {
+		channelPrefix, channelVersionStr, err := splitChannel(startChannel)
+		if err != nil {
+			return nil, err
+		}
+		prefixes = append(prefixes, channelPrefix)
+
+		channelVersion, err := semver.NewVersion(channelVersionStr)
```

Comment:
huh, does this successfully parse two-part versions like 4.16?  I'd have expected it to need be valid SemVer, and [SemVer requires a patch number][1].

[1]: https://semver.org/spec/v2.0.0.html#spec-item-2

Reply:
it works - https://github.com/Masterminds/semver/blob/master/version.go#L177